### PR TITLE
Label graph edges

### PR DESCRIPTION
**Pull request by ttuegel**
@vladciobanu This pull request adds labels to the edges of the execution graph, although it does not actually populate the labels yet. I was thinking that you and I could work on adding that and the user-facing REPL feature to this pull request as an example for @marick of how GitHub allows multiple authors to collaborate on the same request.

Depends on: #524

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by ttuegel** [comment]
I have merged #524 and rebased this pull request onto master. I am going to mark it for review and add the actual labels in a follow-up pull request.

**Comment by ttuegel** [comment]
I have merged #524 and rebased this pull request onto master. I am going to mark it for review and add the actual labels in a follow-up pull request.

File: kore/src/Kore/OnePath/Step.hs [hunk 1]
```diff
@@ -258,41 +258,32 @@ transitionRule
                 -- Filter out ⊥ patterns
                 nonEmptyConfigs = MultiOr.filterOr configs
             if null nonEmptyConfigs
-                then return [(Bottom, proof'')]
-                else return (prove <$> map wrapper (toList nonEmptyConfigs))
+                then return (Bottom, proof'')
+                else Foldable.asum (pure . prove <$> map wrapper (Foldable.toList nonEmptyConfigs))
```
Comment (virgil-serbanuta) [inline kore/src/Kore/OnePath/Step.hs]: Line length.

File: kore/src/Kore/OnePath/Step.hs [hunk 2]
```diff
@@ -336,21 +327,18 @@ transitionRule
                             (MultiOr.extractPatterns remainder)
 
                 if null rewriteResults
-                    then return ((Bottom, combinedProof) : remainderResults)
-                    else return (rewriteResults ++ remainderResults)
+                    then return (Bottom, combinedProof) <|> Foldable.asum (pure <$> remainderResults)
```
Comment (virgil-serbanuta) [inline kore/src/Kore/OnePath/Step.hs]: Line too long (also applies to other places below).

**Pull request merged**